third-party/jemalloc/Makefile

```diff
@@ -12,6 +12,13 @@ ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
 CHPL_JEMALLOC_CFG_OPTIONS += --host=x86_64-cle-linux-gnu
 endif
 
+# Can't build under cce with `-hgnu` or anything above `-O1`. Additionally,
+# CCE doesn't support the dependency flags that jemalloc expects (-MM -MT)
+ifeq ($(CHPL_MAKE_TARGET_COMPILER),cray-prgenv-cray)
+CFLAGS += "-hnognu -O1'
+CHPL_JEMALLOC_MAKE_OPTIONS += CC_MM=''
```

Comment:
You're right that it's not needed. I just wanted to make it more obvious that we're setting it to nothing.